### PR TITLE
MGDOBR-1112: refactor GET bridges/processors api to better specify the filter by prefix name parameters 

### PR DESCRIPTION
**Pull request by r00ta**
MGDOBR-1112

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Review by manstis — CHANGES_REQUESTED** [review]
Filtering by prefix was requested for the UI.

Simply removing it because one persons asks is not correct.

**Comment by r00ta** [comment]
> Filtering by prefix was requested for the UI.
> 
> Simply removing it because one persons asks is not correct.

Yes.. thanks for pointing this out. As per chat this PR (now) aims to improve the api to better specify the parameter name to filter by prefix name

**Comment by r00ta** [comment]
Hi @kelvah , note that we are proposing to change the parameter `name` to `namePrefix` to better specify the behaviour of the api. Let me know if you have any suggestion/concern

File: infra/src/main/java/com/redhat/service/smartevents/infra/api/APIConstants.java [hunk 1]
```diff
@@ -113,9 +113,9 @@ public class APIConstants {
     public static final String SIZE_DEFAULT = "100";
 
     /**
-     * The name query filter parameter name
+     * The name query filter parameter prefix
      */
-    public static final String FILTER_NAME = "name";
+    public static final String PREFIX = "namePrefix";
```
Comment (manstis) [inline infra/src/main/java/com/redhat/service/smartevents/infra/api/APIConstants.java:118]: IMO `FILTER_NAME_PREFIX` would have been nicer as it relates only to the `Name` field.

Making the other parameters `filterNamePrefix` etc.

**Review by manstis — APPROVED** [review]
Nit picking about names; but otherwise fine.

File: infra/src/main/java/com/redhat/service/smartevents/infra/api/APIConstants.java (hunk 1 above)
Comment (r00ta) [reply infra/src/main/java/com/redhat/service/smartevents/infra/api/APIConstants.java]: done

File: manager/src/test/java/com/redhat/service/smartevents/manager/api/user/BridgesAPITest.java [hunk 2]
```diff
@@ -296,7 +296,7 @@ public void testGetBridgesFilterByName() {
         bridge2.setStatus(ACCEPTED);
         bridgeDAO.persist(bridge2);
 
-        BridgeListResponse bridgeListResponse = TestUtils.getBridgesFilterByName(DEFAULT_BRIDGE_NAME + "1").as(BridgeListResponse.class);
+        BridgeListResponse bridgeListResponse = TestUtils.getBridgesFilterByNamePrefix(DEFAULT_BRIDGE_NAME + "1").as(BridgeListResponse.class);
```
Comment (manstis) [inline manager/src/test/java/com/redhat/service/smartevents/manager/api/user/BridgesAPITest.java:299]: If you wanted to be 100% correct you could rename the test too.

File: manager/src/test/java/com/redhat/service/smartevents/manager/api/user/BridgesAPITest.java [hunk 3]
```diff
@@ -393,7 +393,7 @@ public void testGetBridgesFilterByNameAndStatus() {
         bridge2.setStatus(READY);
         bridgeDAO.persist(bridge2);
 
-        BridgeListResponse bridgeListResponse = TestUtils.getBridgesFilterByNameAndStatus(DEFAULT_BRIDGE_NAME + "1", READY).as(BridgeListResponse.class);
+        BridgeListResponse bridgeListResponse = TestUtils.getBridgesFilterByNamePrefixAndStatus(DEFAULT_BRIDGE_NAME + "1", READY).as(BridgeListResponse.class);
```
Comment (manstis) [inline manager/src/test/java/com/redhat/service/smartevents/manager/api/user/BridgesAPITest.java:396]: If you wanted to be 100% correct you could rename the test too.

File: manager/src/test/java/com/redhat/service/smartevents/manager/api/user/ProcessorAPITest.java [hunk 4]
```diff
@@ -125,7 +125,7 @@ public void listProcessorsFilterByName() {
         TestUtils.addProcessorToBridge(bridgeResponse.getId(), new ProcessorRequest("myProcessor", TestUtils.createKafkaAction())).as(ProcessorResponse.class);
         ProcessorResponse p2 = TestUtils.addProcessorToBridge(bridgeResponse.getId(), new ProcessorRequest("myProcessor2", TestUtils.createKafkaAction())).as(ProcessorResponse.class);
 
-        ProcessorListResponse listResponse = TestUtils.listProcessorsFilterByName(bridgeResponse.getId(), "myProcessor2").as(ProcessorListResponse.class);
+        ProcessorListResponse listResponse = TestUtils.listProcessorsFilterByNamePrefix(bridgeResponse.getId(), "myProcessor2").as(ProcessorListResponse.class);
```
Comment (manstis) [inline manager/src/test/java/com/redhat/service/smartevents/manager/api/user/ProcessorAPITest.java:128]: If you wanted to be 100% correct you could rename the test too.

File: manager/src/test/java/com/redhat/service/smartevents/manager/api/user/ProcessorAPITest.java [hunk 5]
```diff
@@ -222,7 +222,7 @@ public void listProcessorsFilterByNameAndStatus() {
         setProcessorStatus(p1.getId(), ACCEPTED);
         setProcessorStatus(p2.getId(), READY);
 
-        ProcessorListResponse listResponse = TestUtils.listProcessorsFilterByNameAndStatus(bridgeResponse.getId(), "myProcessor", READY).as(ProcessorListResponse.class);
+        ProcessorListResponse listResponse = TestUtils.listProcessorsFilterByNamePrefixAndStatus(bridgeResponse.getId(), "myProcessor", READY).as(ProcessorListResponse.class);
```
Comment (manstis) [inline manager/src/test/java/com/redhat/service/smartevents/manager/api/user/ProcessorAPITest.java:225]: If you wanted to be 100% correct you could rename the test too.

File: manager/src/test/java/com/redhat/service/smartevents/manager/api/user/ProcessorAPITest.java [hunk 6]
```diff
@@ -238,7 +238,7 @@ public void listProcessorsFilterByNameAndType() {
         TestUtils.addProcessorToBridge(bridgeResponse.getId(), new ProcessorRequest("myProcessor", TestUtils.createKafkaAction())).as(ProcessorResponse.class);
         ProcessorResponse p2 = TestUtils.addProcessorToBridge(bridgeResponse.getId(), new ProcessorRequest("myProcessor2", TestUtils.createSlackSource())).as(ProcessorResponse.class);
 
-        ProcessorListResponse listResponse = TestUtils.listProcessorsFilterByNameAndType(bridgeResponse.getId(), "myProcessor", SOURCE).as(ProcessorListResponse.class);
+        ProcessorListResponse listResponse = TestUtils.listProcessorsFilterByNamePrefixAndType(bridgeResponse.getId(), "myProcessor", SOURCE).as(ProcessorListResponse.class);
```
Comment (manstis) [inline manager/src/test/java/com/redhat/service/smartevents/manager/api/user/ProcessorAPITest.java:241]: If you wanted to be 100% correct you could rename the test too.

**Review by manstis — APPROVED** [review]
Minor nit picks. Take or leave them. The significant parts of the change are fine.

**Comment by manstis** [comment]
I added @5733d9e2be6485d52ffa08870cabdee0/frontend to the PR for awareness.

@kelvah @vpellegrino @jomarko IDK if you have/want a new JIRA for the FE change?

**Review by sutaakar — APPROVED** [review]

**Comment by robpblake** [comment]
@r00ta @manstis Can't we just document the current behaviour of the `?name` parameter in the OpenAPI spec as providing a potential list and agree that with Heiko as the existing behaviour? Otherwise both Heiko and UI have to change their code for minimal benefit AFAICS.

That would then leave us with Heikos issue MGDOBR-1112 as a feature request for us to correct this. IMHO `?name` should be an exact match and something like `?search` which can take a query-like string for this scenario would be a better option.

Your thoughts?

**Comment by r00ta** [comment]
> @r00ta @manstis Can't we just document the current behaviour of the `?name` parameter in the OpenAPI spec as providing a potential list and agree that with Heiko as the existing behaviour? Otherwise both Heiko and UI have to change their code for minimal benefit AFAICS.
> 
> That would then leave us with Heikos issue MGDOBR-1112 as a feature request for us to correct this. IMHO `?name` should be an exact match and something like `?search` which can take a query-like string for this scenario would be a better option.
> 
> Your thoughts?

@robpblake up to you. Feel free to update the jira with the desired behaviour and I'll implement it

**Pull request closed without merging**